src/Util/Hash.h

```diff
@@ -0,0 +1,24 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2020 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include <cstddef>
+#include <functional>
+
+namespace openblack
+{
+// https://www.boost.org/doc/libs/1_70_0/doc/html/hash/reference.html#boost.hash_combine
+template <class T>
+inline void hash_combine(std::size_t& seed, const T& v)
+{
+	std::hash<T> hasher;
+	seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
```

Comment:
Why not use `entt::cache` for this? https://github.com/skypjack/entt/wiki/Crash-Course:-resource-management